examples/tutorials/text.py

```diff
@@ -0,0 +1,128 @@
+"""
+Plotting text
+=============
+
+It is often useful to add annotations to a map plot. This is handled by
+:meth:`pygmt.Figure.text`.
+"""
+
+import pygmt
+
+########################################################################################
+# Basic map annotation
+# --------------------
+#
+# Text annotations can be added to a map using the :meth:`text` method of the
+# :class:`pygmt.Figure`.
+#
+# Full details of the GMT6 command `text` can be found `here<https://docs.generic-mapping-tools.org/latest/text.html>`_.
+# The Python binding to this command is documented `here<https://www.pygmt.org/latest/api/generated/pygmt.Figure.text.html>`_.
+#
+# Here we create a simple map and add an annotation using the ``text``, ``x``,
+# and ``y`` arguments to specify the annotation text and position in the
+# projection frame. ``text``, ``x``, and ``y`` accept `int`, `str`, or `float`.
+
+fig = pygmt.Figure()
+with pygmt.config(MAP_FRAME_TYPE="plain"):
+    fig.basemap(region=[108, 120, -5, 8], projection="M20c", frame="a")
+fig.coast(land="black", water="skyblue")
+
+# Plotting text annotations using single elements
+fig.text(text="SOUTH CHINA SEA", x=112, y=6)
+
+# Plotting text annotations using lists of elements
+fig.text(text=["CELEBES SEA", "JAVA SEA"], x=[119, 112], y=[3.25, -4.6])
+
+fig.show()
+
+########################################################################################
+# Changing font style
+# -------------------
+# The size, family/weight, and colour of an annotation can be specified using the ``font`` argument.
+#
+# A list of all recognised fonts can be found `here<https://docs.generic-mapping-tools.org/latest/cookbook/postscript_fonts.html>`_),
+# including details of how to use non-default fonts.
+
+fig = pygmt.Figure()
+with pygmt.config(MAP_FRAME_TYPE="plain"):
+    fig.basemap(region=[108, 120, -5, 8], projection="M20c", frame="a")
+fig.coast(land="black", water="skyblue")
+
+# Customising the font style
+fig.text(text="BORNEO", x="114.", y=0.5, font="22p,Helvetica-Bold,white")
+
+fig.show()
+
+########################################################################################
+# Plotting from a text file
+# -------------------------
+#
+# It is also possible to add annotations from a file containing `x`, `y`, and
+# `text` fields. Here we give a complete example.
+
+fig = pygmt.Figure()
+with pygmt.config(MAP_FRAME_TYPE="plain"):
+    fig.basemap(region=[108, 120, -5, 8], projection="M20c", frame="a")
+fig.coast(land="black", water="skyblue")
+
+# Plot region names / sea names
+fig.text(textfiles="examples.txt", font="22p,Helvetica-Bold,white")
```

Comment:
It would be nice to be able to show what the contents of the text file looks like. There's probably a quick way to generate a text file directly with python?

Reply:
Yeah, something like:

```python
with open('examples.txt', 'w') as f:
    print('114,0.5,BORNEO.', file=f)
```

I did wonder about that. Is it worth including something to wipe the file at the end or just happy to leave it?

Reply:
Ah cool, I didn't know you could `print()` to a file like that! Learn something new everyday :smile:

> I did wonder about that. Is it worth including something to wipe the file at the end or just happy to leave it?

Maybe try using [`tempfile`](https://docs.python.org/3/library/tempfile.html)? See https://stackoverflow.com/questions/5344287/create-read-from-tempfile for an example. Something like:

```python
with tempfile.NamedTemporaryFile():
    fig.text(...)
```

Reply:
I personally prefer to codes like:
```
with open('examples.txt', 'w') as f:
    f.write('114 0.5 BORNEO.')
```
since `f.write()` is more commonly used than `print("xxx", file=f)` and `tempfile`, and more readable to new Python users.

I also prefer to have the input data space-separated. Although GMT supports many separators, e.g., spaces, tabs, and commas, commas are also used as the separator of font attributes, e.g., `22p,Helvetica-Bold,white`.

Reply:
Sure, in defense of `print(..., file=f)`, it calls str() on whatever is passed in as a list so you can do `print(114, 0.5, "BORNEO", file=f). However, you're right that explicit is better than implicit, so:

```python

# Create file
with open("examples.txt", "w") as f:
    f.write("114 0.5 BORNEO")

# Plot region names / sea names
fig.text(textfiles="examples.txt", font="22p,Helvetica-Bold,white")

# Cleanup
os.remove("examples.txt")
```

Reply:
Ok, as per your code at https://github.com/GenericMappingTools/pygmt/pull/480/files#r443139292, I've updated the code at commit 20994aa0294dec11ca376dd4c2fac485fdb04e42